modules/renderer/src/text-measurer.ts

```diff
@@ -34,17 +34,19 @@ export default class TextMeasurer {
     this.ctx.font = `${fontStyle} ${fontWeight} ${fontSize}px ${fontFamily}`
   }
 
-  measure(text: string): Size {
-    const {fontSize, lineHeight} = this.opts
-    const rowHeight = fontSize * 1.2
-    const rowSpacing = fontSize * (lineHeight - 1)
+  measure(text: string, fontSize: number, fontFamily: string, fontStyle: string): Size {
     let w = 0
+    let h = 0
     const lines = text.split('\n')
+    this.ctx.font = fontStyle + ' ' + fontSize.toString() + ' ' + fontFamily
+    let lineH: number
     for (const line of lines) {
       const metrics = this.ctx.measureText(line)
       w = Math.max(w, metrics.width)
+      lineH = metrics.actualBoundingBoxAscent
+      h += 2 * lineH
```

Comment:
What is 2? There is a `lineHeight` user option.

Reply:
I have experimented with it. It seems the blank space in between two lines is close to the same value, so with every line the height grows twice the value, except the last line.

Reply:
That's because you are only measuring the vertical distance between the bounding box top and the baseline. Doesn't justify you discarding a value that the user has provided.

Reply:
Which value do you reference that is ignored?

Reply:
All method parameters are used in this line https://github.com/microsoft/msagljs/blob/svgout/modules/renderer/src/text-measurer.ts#:~:text=this.ctx.font%20%3D%20fontStyle%20%2B%20%27%20%27%20%2B%20fontSize.toString()%20%2B%20%27%20%27%20%2B%20fontFamily

Reply:
Changing the height calculation as ` h += metrics.actualBoundingBoxAscent + metrics.actualBoundingBoxDescent`. The results are still not perfect, but better than before.